### PR TITLE
feat: adding addPulse functionality

### PR DESCRIPTION
**Pull request by MinaZakaria**
Adding addPulse functionality to open the cash drawer

related to the issue: https://github.com/tr3v3r/react-native-esc-pos-printer/issues/17 

Tested on IOS Simulator
Tested on Android Device Version 11

### CODE REVIEW

**Comment by tr3v3r** [comment]
@MinaZakaria hi! Thanks!
Looks fine. One moment, could you please update docs as well?

File: android/src/main/java/com/reactnativeescposprinter/EscPosPrinterModule.java [hunk 1]
```diff
@@ -170,6 +172,7 @@ public String getName() {
       constants.put("EPOS2_SYMBOL_QRCODE_MODEL_1", Printer.SYMBOL_QRCODE_MODEL_1);
       constants.put("EPOS2_SYMBOL_QRCODE_MODEL_2", Printer.SYMBOL_QRCODE_MODEL_2);
       constants.put("EPOS2_SYMBOL_QRCODE_MICRO", Printer.SYMBOL_QRCODE_MICRO);
+      constants.put("EPOS2_PARAM_DEFAULT", Printer.PARAM_DEFAULT);
```
Comment (tr3v3r) [inline android/src/main/java/com/reactnativeescposprinter/EscPosPrinterModule.java:175]: Do we need this constant?

**Review by tr3v3r — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/reactnativeescposprinter/EscPosPrinterModule.java (hunk 1 above)
Comment (MinaZakaria) [reply android/src/main/java/com/reactnativeescposprinter/EscPosPrinterModule.java:175]: no we don't .. I removed it

**Comment by MinaZakaria** [comment]
Could you please merge the feature if there is no any further comments ?

**Comment by tr3v3r** [comment]
@MinaZakaria hi! do you have time to resolve the latest comment? (About pin parameter)

**Comment by MinaZakaria** [comment]
> @MinaZakaria hi! do you have time to resolve the latest comment? (About pin parameter)

I think you forget to submit the latest comment .. I cannot find it

**Comment by tr3v3r** [comment]
@MinaZakaria sorry for that( just submitted. After update I’ll immediately merge and prepare new release

**Comment by MinaZakaria** [comment]
@tr3v3r  The latest comment applied but I didn't test as I cannot access the drawer right now. You can review and test if you want. Or I'll test it next Sunday.

**Comment by tr3v3r** [comment]
@MinaZakaria actually your code will not work because you need use numbers exporter from native. ( see image printing method as example ). 

Ok. If I have a capacity I’ll update it by my own tomorrow

**Comment by MinaZakaria** [comment]
Thank you for merging .. could you please add a new release or should I use the main branch?

**Comment by tr3v3r** [comment]
@MinaZakaria hi. Release is planned for tomorrow

**Comment by tr3v3r** [comment]
@MinaZakaria  just released version 1.5.0. Please check if everything works as expected

**Comment by MinaZakaria** [comment]
Hello @tr3v3r 
Unfortunately. it's not working.
the assertion failed with the error
```
[Error: Error: EPOS2_DRAWER_2PIN is not valide property value for addPulse]
```
same error if used `EPOS2_DRAWER_5PIN`

when I print `EscPosPrinter.getConstants();` It doesn't have the `EPOS2_DRAWER_2PIN` key !
I don't know what is the cause of the problem.

**Comment by tr3v3r** [comment]
@MinaZakaria have you run pod install after update ?

**Comment by tr3v3r** [comment]
@MinaZakaria Just tested on iOS on version 1.5.0 - no errors. 
Could you please try to clean and re-install the latest version and confirm one more time?

**Comment by MinaZakaria** [comment]
Just tested and everything is working like a Charm. Thank you 👍 💘

**Pull request merged**